pkg/gcp/actuator/actuator.go

```diff
@@ -238,6 +286,59 @@ func (a *Actuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest) er
 	return nil
 }
 
+// syncSTSSecret makes a time-based token available in a Secret in the namespace of an operator that
+// has supplied the following in the CredentialsRequest:
+// a non-empty cr.CloudTokenPath
+// a non-empty cr.ProviderSpec.ServiceAccountEmail
+// a non-empty cr.ProviderSpec.PoolID
+// a non-empty cr.ProviderSpec.ProviderID
+func (a *Actuator) syncSTSSecret(poolName, providerId, serviceAccount, cloudTokenPath string, cr *minterv1.CredentialsRequest, logger log.FieldLogger, ctx context.Context) error {
+	gcpClient, err := a.buildRootGCPClient(cr)
```

Comment:
I don't think we can assume STS clusters have the Root client... they probably have `buildReadGCPClient`, though.

Reply:
Actually, can this be derived from the PoolID or ProviderID? That would be more accurate to what it is being used for.

Reply:
Even so, you would still need to `buildReadGCPClient`... and that client probably doesn't have the permissions to check the PoolID or ProviderID. It might be best to revert to asking for the ProjectID like you originally did.

Reply:
Alternatively, you could ask for the audience directly instead of computing it based on multiple inputs.